geode-core/src/test/java/com/gemstone/gemfire/cache/query/QueryWithBucketParameterIntgTest.java

```diff
@@ -0,0 +1,224 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * QueryWithBucketParameterTest.java
+ * JUnit based test
+ *
+ * Created on April 13, 2005, 2:40 PM
+ */
+package com.gemstone.gemfire.cache.query;
+
+        import static org.junit.Assert.*;
+
+        import java.io.Serializable;
+        import java.util.*;
+
+        import org.junit.After;
+        import org.junit.Before;
+        import org.junit.Rule;
+        import org.junit.Test;
+        import org.junit.experimental.categories.Category;
+
+        import com.gemstone.gemfire.cache.*;
+        import com.gemstone.gemfire.cache.query.internal.DefaultQuery;
+        import com.gemstone.gemfire.internal.cache.LocalDataSet;
+        import com.gemstone.gemfire.internal.cache.PartitionedRegion;
+        import com.gemstone.gemfire.test.junit.categories.IntegrationTest;
+        import org.junit.rules.ExpectedException;
+
+/**
+ * @implNote This test class has been designed to test the query execution using the LocalDataSet.executeQuery
+ * where one of the parameters passed is the bucket set to be used.
+ * Different variation of hashset variables are passed to the function to check for errors.
+ * Created by nnag on 3/3/16.
+ */
+@Category(IntegrationTest.class)
+public class QueryWithBucketParameterIntgTest {
+
+    //Necessary data elements required for execution of the test
+//    Region region;
+//    QueryService qs;
+//    Cache cache;
+    DefaultQuery queryExecutor;
+    LocalDataSet lds;
+
+    /**
+     * @implNote This code is copied from BugJUnitTest.java. These elements are required to pass as parameters
+     * to create data elements for the region
+     * @implNote This needs to be moved to a common test package to avoid redundant code.
+     */
+    static class MyValue implements Serializable, Comparable<MyValue>
+    {
+        public int value = 0;
+
+        public MyValue(int value) {
+            this.value = value;
+        }
+
+
+        public int compareTo(MyValue o)
+        {
+            if(this.value > o.value) {
+                return 1;
+            }else if(this.value < o.value) {
+                return -1;
+            }else {
+                return 0;
+            }
+        }
+    }
+
+    //Empty constructor
+    public QueryWithBucketParameterIntgTest() {
+    }
+
+    /**
+     * Pre-test execution to create the test bed.
+     * This code snippet has been borrowed from (com.gemstone.gemfire.cache.query) BugJUintTest.java [03/02/2016]
```

Comment:
Don't think you need this disclaimer as well as the comment as the tag setup is self explanatory